Wikipedia/Code/WMFModalImageGalleryViewController.m

```diff
@@ -478,6 +525,13 @@ - (void)updateImageForCell:(WMFImageGalleryCollectionViewCell*)cell
         [self setPlaceholderImage:download.image ofInfo:infoForImage forCellAtIndexPath:indexPath];
     }]
     .catch(^(NSError* error) {
+        @strongify(self);
+        BOOL const isInvalidURL =
```

Comment:
@coreyfloyd ah, i know why: we use the same code path for setting gallery placeholders regardless of whether the image info is ready, which results in the image info being `nil`, which results in the high-res being `nil`, which results in the banner (invalid/empty URL).